GraphsV2/LeafletGraph/index.js

```diff
@@ -0,0 +1,240 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import { Map, TileLayer, Marker, Tooltip } from "react-leaflet";
+import isEqual from 'lodash/isEqual';
+import MarkerClusterer from "react-leaflet-markercluster";
+import { compose } from 'redux';
+import 'react-leaflet-markercluster/dist/styles.min.css';
+import L from 'leaflet';
+
+import WithConfigHOC from '../../HOC/WithConfigHOC';
+import WithValidationHOC from '../../HOC/WithValidationHOC';
+import SearchBar from "../../SearchBar";
+import { getIconPath } from '../../utils/helpers';
+import { config } from './default.config';
+import { styled } from '@material-ui/core/styles';
+
+//let markers = new Map();
+let timerId = null;
+
+const InfoContainer = styled('div')({
+  display: 'table'
+});
+
+const InfoBox = styled('div')({
+  display: 'table-row'
+});
+
+const Label = styled('small')({
+  fontSize: 9,
+  color: 'rgb(106, 106, 106)',
+  whiteSpace: 'nowrap',
+  display: 'table-cell',
+  marginRight: '10px'
+});
+
+const Data = styled('small')({
+  whiteSpace: 'nowrap',
+  display: 'table-cell',
+  marginLeft: '10px'
+});
+
+// call on marker click
+const handleMarkerClick = (data, onMarkClick) => {
+  return onMarkClick ? onMarkClick(data) : "";
+}
+
+const LeafletGraph = (props) => {
+
+  const [stateData, setStateData] = useState([]);
+  const [infowindow, setInfowindow] = useState({ data: null, position: null });
+
+  useEffect(() => {
+    initiate(props);
+
+    return () => {
+      clearTimeout(timerId);
+    }
+  }, [props.data]);
+
+  const {
+    data,
+    width,
+    properties,
+    onMarkClick,
+  } = props;
+
+  let { 
+    height,
+  } = props;
+
+  const {
+    searchBar,
+    maxZoom,
+    minZoom,
+    latitudeColumn,
+    longitudeColumn,
+    localityColumn,
+    nameColumn,
+    criticalAlarmColumn,
+    majorAlarmColumn,
+    minorAlarmColumn,
+    bootstrapStatusColumn,
+    NSGVersionColumn,
+    idColumn,
+    markerIcon,
+    filters,
+  } = properties;
+
+  const initiate = (props) => {
+    //markers = new Map();
+    setStateData(props.data);
+    setInfowindow({ data: null, position: null });
+  }
+
+  // toggle info window on marker click
+  const toggleInfoWindow = (data = null, position = null) => {
+    if (!data || !isEqual(position, infowindow.position)) {
+      setInfowindow({ data, position });
+    }
+  }
+
+  const displayNSGInfo = (data) => {
+    const displayInfo = info => {
+      return info.map((row, i) => (
+        <InfoBox key={i}>
+          <Label>{row.label}</Label>
+          <Data>&nbsp;</Data>
+          <Data>{row.text}</Data>
+        </InfoBox>
+      ))
+    }
+
+    return (
+      <InfoContainer className={'geoGraph'}>
+        {
+          displayInfo([
```

Comment:
This is specific to NSGs however I think this should be generic so we can add any objects with a location. We should define this in the vis config file similarly how we define the table columns. Let's take this opportunity to improve on the framework.